plugin/src/main/java/org/opensearch/ml/helper/ConnectorAccessControlHelper.java

```diff
@@ -100,11 +101,11 @@ public void getConnector(Client client, String connectorId, ActionListener<Conne
                     listener.onFailure(e);
                 }
             } else {
-                listener.onFailure(new MLResourceNotFoundException("Fail to find connector:" + connectorId));
+                listener.onFailure(new OpenSearchStatusException("Failed to find connector:" + connectorId, RestStatus.NOT_FOUND));
             }
         }, e -> {
-            log.error("Fail to get connector", e);
-            listener.onFailure(new IllegalStateException("Fail to get connector:" + connectorId));
+            log.error("Failed to get connector", e);
+            listener.onFailure(new IllegalStateException("Failed to get connector:" + connectorId));
```

Comment:
Don't we need `OpenSearchStatusException`  here too?

Reply:
Changed